src/Sarif/Core/Stack.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Text;
+using DotNetStackFrame = System.Diagnostics.StackFrame;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public enum StackFormat
+    {
+        Default,
+        TrailingNewLine
+    }
+
+    /// <summary>
+    /// A call stack relevant to a SARIF result.
+    /// </summary>
+    public partial class Stack
+    {
+        private const string IN = " in ";
+        private const string AT = "   at ";
+        private const string LINE = ":line";
+
+        /// <summary>
+        /// Creates a SARIF Stack instance from a .NET StackTrace instance
+        /// </summary>
+        /// <param name="stackTrace"></param>
+        /// <returns></returns>
+        public static Stack Create(StackTrace stackTrace)
+        {
+            Stack stack = new Stack();
+
+            if (stackTrace.FrameCount == 0) { return stack; }
+
+            stack.Frames = new StackFrame[stackTrace.FrameCount];
+
+            for (int i = 0; i < stackTrace.FrameCount; i++)
+            {
+                stack.Frames[i] = StackFrame.Create(stackTrace.GetFrame(i));                
+            }
+            return stack;
+        }
+
+        public static IEnumerable<Stack> Create(Exception exception)
+        {
+            List<Stack> stacks;
+            Queue<Exception> exceptions;
+
+            stacks = new List<Stack>();
+            exceptions = new Queue<Exception>(new Exception[] { exception });
```

Comment:
A good question and one that will come up again when you examine the StackTrace line parsing code (which creates many strings rather than parsing the line character by character as could be done).

I favored readability, assuming as you allude to that these small inefficiencies aren't worth tuning for, given the relative rarity of these code paths. Otoh, if these are easily unit-tested methods (as they are) and readability is not at great risk (it probably is not), why not author the most efficient implementation possible? Glad to hear your thoughts and Larry's.